testsuite/integration/elytron-oidc-client/src/test/java/org/wildfly/test/integration/elytron/oidc/client/logout/OidcLogoutBaseTest.java

```diff
@@ -0,0 +1,521 @@
+    public void testRpInitiatedLogout() throws Exception {
+
+        loginToApp(RP_INITIATED_LOGOUT_APP);
+        assertUserLoggedIn(RP_INITIATED_LOGOUT_APP, "GOOD");
+        logoutOfKeycloak(RP_INITIATED_LOGOUT_APP, "You are logged out");
+        assertUserLoggedOut(RP_INITIATED_LOGOUT_APP, "GOOD");
+    }
+
+    @Test
+    @OperateOnDeployment(FRONT_CHANNEL_LOGOUT_APP)
+    public void testFrontChannelLogout() throws Exception {
+
+        loginToApp(FRONT_CHANNEL_LOGOUT_APP);
+        assertUserLoggedIn(FRONT_CHANNEL_LOGOUT_APP, "GOOD");
+        logoutOfKeycloak(FRONT_CHANNEL_LOGOUT_APP, "You are logging out from following apps");
+        assertUserLoggedOut(FRONT_CHANNEL_LOGOUT_APP, "GOOD");
+    }
+
+    @Test
+    @OperateOnDeployment(BACK_CHANNEL_LOGOUT_APP)
+    public void testBackChannelLogout() throws Exception {
+
+        loginToApp(BACK_CHANNEL_LOGOUT_APP);
+        assertUserLoggedIn(BACK_CHANNEL_LOGOUT_APP, "GOOD");
+        logoutOfKeycloak(BACK_CHANNEL_LOGOUT_APP,"You are logged out");
+        assertUserLoggedOut(BACK_CHANNEL_LOGOUT_APP, "Sign in to your account");
+    }
+
+    @Test
+    @OperateOnDeployment(POST_LOGOUT_APP)
+    public void testPostLogout() throws Exception {
+
+        loginToApp(POST_LOGOUT_APP);
+        assertUserLoggedIn(POST_LOGOUT_APP, "GOOD");
+        logoutOfKeycloak(POST_LOGOUT_APP, "You are logged out");
+        assertUserLoggedOut(POST_LOGOUT_APP, "GOOD");
+    }
+
+    @Test
+    public void testBackChannelLogoutTwo() throws Exception {
+        loginToApp(BACK_CHANNEL_LOGOUT_APP);
+        loginToApp(FRONT_CHANNEL_LOGOUT_APP);
+        assertUserLoggedIn(BACK_CHANNEL_LOGOUT_APP, "GOOD");
+        assertUserLoggedIn(FRONT_CHANNEL_LOGOUT_APP, "GOOD");
+        logoutOfKeycloak(BACK_CHANNEL_LOGOUT_APP,"You are logged out");
+        assertUserLoggedOut(BACK_CHANNEL_LOGOUT_APP, "Sign in to your account");
+        assertUserLoggedOut(FRONT_CHANNEL_LOGOUT_APP, "Sign in to your account");
+    }
+
+    private static URL generateURL(String appName) {
+        try {
+            return new URL("http", TestSuiteEnvironment.getHttpAddress(),
+                    TestSuiteEnvironment.getHttpPort(),
+                    "/" + appName + SimpleSecuredServlet.SERVLET_PATH);
+        } catch (MalformedURLException e) {
+            assertFalse(e.getMessage(), false);
+        }
+        return null;
+    }
+
+    public static void loginToApp(String appName) throws Exception {
+        loginToApp(appName, ALICE, ALICE_PASSWORD, HttpURLConnection.HTTP_OK,
+                SimpleServlet.RESPONSE_BODY);
+    }
+
+    public static void loginToApp(String appName,
+                                  String username, String password, int expectedStatusCode, String expectedText) throws Exception {
+        loginToApp(username, password, expectedStatusCode, expectedText, true,
+                generateURL(appName).toURI());
+    }
+
+    public static void loginToApp(String username, String password,
```

Comment:
Just to check, I think there's a similar method in OidcBaseTest, right? Can we use a common method implementation for both?